x-pack/solutions/observability/plugins/observability_agent/server/tools/get_change_points/get_metric_change_points.ts

```diff
@@ -0,0 +1,334 @@
+      index,
+      size: 0,
+      track_total_hits: false,
+      query: {
+        bool: {
+          filter: filters,
+        },
+      },
+      aggs: {
+        groups: {
+          ...groupAgg,
+          aggs: subAggs,
+        },
+      },
+    });
+
+    const groups = (
+      response.aggregations?.groups && 'buckets' in response.aggregations.groups
+        ? response.aggregations.groups.buckets || []
+        : []
+    ) as Array<AggregateOfMap<typeof subAggs, unknown> & { key?: string; key_as_string?: string }>;
+
+    const series = groups.reduce((acc, group) => {
+      const key = group.key ?? 'all';
+
+      const changes = group.changes as AggregateOf<
+        { change_point: { buckets_path: string } },
+        unknown
+      >;
+
+      const [changeType, value] = Object.entries(changes.type)[0];
+
+      // filter out indeterminable changes
+      if (changes.type.indeterminable || !changes.bucket?.key) {
+        return acc;
+      }
+
+      const changePoint = {
+        key,
+        over_time: group.over_time.buckets.map((bucket) => {
+          return {
+            x: new Date(bucket.key_as_string).getTime(),
+            y: bucket.value?.value as number | null,
+          };
+        }),
+        changes: {
+          time: new Date(changes.bucket.key).toISOString(),
+          type: changeType,
+          ...value,
+        },
+      };
+
+      return [...acc, changePoint];
+    }, [] as ChangePoint[]);
+
+    return series;
+  } catch (error) {
+    throw error;
+  }
+}
+
+const getMetricChangePointsSchema = z.object({
+  start: z
+    .string()
+    .describe(
+      'The beginning of the time range, in Elasticsearch datemath, like `now-24h`, or an ISO timestamp'
+    ),
+  end: z
+    .string()
+    .describe(
+      'The end of the time range, in Elasticsearch datemath, like `now`, or an ISO timestamp'
+    ),
+  metrics: z
+    .array(
+      z.object({
+        name: z.string().describe('The name of the set of metrics'),
+        index: z.string().describe('The index or index pattern to find the metrics').optional(),
+        kqlFilter: z
+          .string()
+          .describe('A KQL filter to filter the metric documents, e.g.: my_field:foo')
+          .optional(),
+        field: z
+          .string()
+          .describe(
+            'Metric field that contains the metric. Only use if the metric aggregation type is not `count`.'
+          )
+          .optional(),
+        type: z
+          .enum(['count', 'avg', 'sum', 'min', 'max', 'p95', 'p99'])
+          .describe('The type of metric aggregation to perform. Defaults to `count`')
+          .optional(),
+        groupBy: z
+          .array(z.string())
+          .describe('Optional keyword fields to group metrics by.')
+          .optional(),
+      })
+    )
+    .describe(
+      'Analyze changes in metrics. DO NOT UNDER ANY CIRCUMSTANCES use date or metric fields for groupBy. Leave empty unless needed.'
```

Comment:
Shouldn't the comment about `groupBy` be moved to the `groupBy` description?

```suggestion
      'Analyze changes in metrics'
```

Reply:
The description for the metric field could then be:

```
Optional keyword fields to group metrics by. DO NOT UNDER ANY CIRCUMSTANCES use high cardinality fields like date or metric fields. Leave empty unless needed.
```

Reply:
replaced